src/net_processing.cpp

```diff
@@ -271,12 +271,15 @@ struct Peer {
 
     struct TxRelay {
         mutable RecursiveMutex m_bloom_filter_mutex;
-        /** Whether the peer wishes to receive transaction announcements.
+        /** Whether we relay transactions to this peer.
          *
-         * This is initially set based on the fRelay flag in the received
-         * `version` message. If initially set to false, it can only be flipped
-         * to true if we have offered the peer NODE_BLOOM services and it sends
-         * us a `filterload` or `filterclear` message. See BIP37. */
+         * On receiving a `version` message, this is initially set to false
+         * if the peer is outbound block-relay-only and otherwise based on
+         * the fRelay flag.
+         *
+         * If initially set to false, it can only be flipped to true if we have
+         * offered the peer NODE_BLOOM services and it sends us a `filterload`
+         * or `filterclear` message. See BIP37. */
```

Comment:
This sentence could need an update after #22778: If we haven't offered `NODE_BLOOM` and get `fRelay=false`, we no longer set `m_relay_txs` to false, but now don't initialise a TxRelay object anymore.
So the case where it is initialised to `false` and is impossible to flip cannot happen anymore.

Reply:
Thanks, removed per https://github.com/bitcoin/bitcoin/pull/26109/files#r975018566.